plugin_tests/submission_test.py

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import json
+import mock
+
+from tests import base
+
+
+def setUpModule():
+    base.enabledPlugins.append('covalic')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class SubmissionBase(base.TestCase):
+    def setUp(self):
+        super(SubmissionBase, self).setUp()
+
+        from girder.plugins.jobs.models.job import Job
+        from girder.plugins.worker.constants import PluginSettings as WorkerSettings
+        from girder.plugins.covalic.constants import PluginSettings as CovalicSettings
+
+        self.scheduleJobMock = mock.patch.object(Job, 'scheduleJob')
+        self.scheduleJobMock.start()
+
+        self.model('setting').set(WorkerSettings.API_URL, 'http://example.com/api/v1')
+
+        user = {
+            'email': 'admin@email.com',
+            'login': 'amdinlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'adminpassword',
+            'admin': True
+        }
+        self.admin = self.model('user').createUser(**user)
+
+        user = {
+            'email': 'good@email.com',
+            'login': 'goodlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'goodpassword',
+            'admin': False
+        }
+        self.user = self.model('user').createUser(**user)
+
+        self.group = self.model('group').createGroup('phase1', self.admin)
+        self.model('group').addUser(self.group, self.user)
+
+        user = {
+            'email': 'scoring@email.com',
+            'login': 'scoringlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'scoringpassword',
+            'admin': False
+        }
+        self.scoringUser = self.model('user').createUser(**user)
+        self.model('setting').set(
+            CovalicSettings.SCORING_USER_ID,
+            self.scoringUser['_id']
+        )
+
+        self.challenge = self.model('challenge', 'covalic').createChallenge(
+            name='challenge 1',
+            creator=self.admin,
+            public=False)
+
+        metrics = {
+            'accuracy': {'weight': 1}
```

Comment:
It would be good to test with more than one metric. Could you add another and assign them different weights?